fcrepo-kernel-api/src/test/java/org/fcrepo/kernel/api/services/functions/ConfigurableHierarchicalSupplierTest.java

```diff
@@ -39,11 +50,11 @@ public void testGetIdNoPairtree() {
 
     @Test
     public void testGetIdPairtreeParams() {
-        final UniqueValueSupplier defaultPidMinter = new ConfigurableHierarchicalSupplier(2, 4);
+        final UniqueValueSupplier defaultPidMinter = new ConfigurableHierarchicalSupplier();
         final String id = defaultPidMinter.get();
         // With (desiredLength > 0 && desiredCount > 0) check to see that id contains (count + 1) parts
         final int parts = (id.split("/").length);
         assertEquals(5, parts);
     }
 
-}
\ No newline at end of file
+}
```

Comment:
So this test can be done using `setField` which allows you to "Inject" the class, but still change it in the test.
```java
@RunWith(MockitoJUnitRunner.Silent.class)
public class ConfigurableHierarchicalSupplierTest {

    private final FedoraPropsConfig propsConfig = new FedoraPropsConfig();

    private final UniqueValueSupplier defaultPidMinter = new ConfigurableHierarchicalSupplier();

    @Before
    public void setUp() {
        // Need to set the defaults 
        propsConfig.setFcrepoPidMinterLength(0);
        propsConfig.setFcrepoPidMinterCount(0);
        setField(defaultPidMinter, "fedoraPropsConfig", propsConfig);
    }

    @Test
    public void testGet() {
        final String id = defaultPidMinter.get();
        // No pairtrees is default; with no args check to see that id contains just the 1 pid part
        final int parts = (id.split("/").length);
        assertEquals(1, parts);
    }

    @Test
    public void testGetIdNoPairtree() {
        final String id = defaultPidMinter.get();
        // With (desiredLength,desiredCount=0), check to see that id contains 1 part and no slashes
        final int parts = (id.split("/").length);
        assertEquals(1, parts);
    }

    @Test
    public void testGetIdPairtreeParams() {
        // Alter the settings for this test.
        propsConfig.setFcrepoPidMinterLength(2);
        propsConfig.setFcrepoPidMinterCount(4);
        final String id = defaultPidMinter.get();
        // With (desiredLength > 0 && desiredCount > 0) check to see that id contains (count + 1) parts
        final int parts = (id.split("/").length);
        assertEquals(5, parts);
    }
}
```

Reply:
I've made the recommended changes but I'm back to getting an NPE on the ``` final String id = defaultPidMinter.get();``` liens in each test